src/geometry/euclidean.lean

```diff
@@ -33,3 +35,634 @@ normed_add_torsor V P
+                                         (angle_of_vectors_le_pi _ _)
+                                         (angle_of_vectors_nonneg _ _)
+                                         (angle_of_vectors_le_pi _ _) _,
+  rw [cos_angle_of_vectors, cos_angle_of_vectors, h, ←neg_sub, norm_neg, neg_sub,
+      inner_sub_right, inner_sub_right, inner_self_eq_norm_square, inner_self_eq_norm_square, h,
+      inner_comm x y]
+end
+
+/-- Converse of pons asinorum, vector angle form. -/
+lemma norm_eq_of_angle_sub_eq_angle_sub_rev_of_angle_ne_pi {x y : V}
+    (h : angle_of_vectors x (x - y) = angle_of_vectors y (y - x))
+    (hpi : angle_of_vectors x y ≠ real.pi) : ∥x∥ = ∥y∥ :=
+begin
+  replace h := real.arccos_inj (abs_le.mp (abs_inner_div_norm_mul_norm_le_one x (x - y))).1
+                               (abs_le.mp (abs_inner_div_norm_mul_norm_le_one x (x - y))).2
+                               (abs_le.mp (abs_inner_div_norm_mul_norm_le_one y (y - x))).1
+                               (abs_le.mp (abs_inner_div_norm_mul_norm_le_one y (y - x))).2
+                               h,
+  by_cases hxy : x = y,
+  { rw hxy },
+  { rw [←norm_neg (y - x), neg_sub, mul_comm, mul_comm ∥y∥, div_eq_mul_inv, div_eq_mul_inv,
+        mul_inv', mul_inv', ←mul_assoc, ←mul_assoc] at h,
+    replace h :=
+      mul_right_cancel' (inv_ne_zero (λ hz, hxy (eq_of_sub_eq_zero (norm_eq_zero.1 hz)))) h,
+    rw [inner_sub_right, inner_sub_right, inner_comm y x, inner_self_eq_norm_square,
+        inner_self_eq_norm_square, mul_sub_right_distrib, mul_sub_right_distrib,
+        mul_self_mul_inv, mul_self_mul_inv, sub_eq_sub_iff_sub_eq_sub,
+        ←mul_sub_left_distrib] at h,
+    by_cases hx0 : x = 0,
+    { rw [hx0, norm_zero, inner_zero_left, zero_mul, zero_sub, neg_eq_zero] at h,
+      rw [hx0, norm_zero, h] },
+    { by_cases hy0 : y = 0,
+      { rw [hy0, norm_zero, inner_zero_right, zero_mul, sub_zero] at h,
+        rw [hy0, norm_zero, h] },
+      { rw [inv_sub_inv (λ hz, hx0 (norm_eq_zero.1 hz)) (λ hz, hy0 (norm_eq_zero.1 hz)),
+            ←neg_sub, ←mul_div_assoc, mul_comm, mul_div_assoc, ←mul_neg_one] at h,
+        symmetry,
+        by_contradiction hyx,
+        replace h := (mul_left_cancel' (sub_ne_zero_of_ne hyx) h).symm,
+        rw [inner_div_norm_mul_norm_eq_neg_one_iff, ←angle_of_vectors_eq_pi_iff] at h,
+        exact hpi h } } }
+end
+
+end real_inner_product
+
+section euclidean
+/-!
+### Geometrical results on Euclidean affine spaces
+
+This section develops some geometrical definitions and results on
+Euclidean affine spaces.
+-/
+
+open add_torsor
+
+variables (V : Type*) {P : Type*} [inner_product_space V] [metric_space P]
+    [euclidean_affine_space V P]
+include V
+
+/-- The undirected angle at `p2` between the line segments to `p1` and
+`p3`. If either of those points equals `p2`, this is π/2. -/
+def angle_of_points (p1 p2 p3 : P) : ℝ := angle_of_vectors (p1 -ᵥ p2 : V) (p3 -ᵥ p2)
+
+notation `∠` := angle_of_points
```

Comment:
I don't think this should be global notation.
You could add `local ` in front of this line, to restrict it to this file.
Alternatively, you can make it `localized`, so that writing `open_locale euclidean_geometry` (or something) will enable the notation.

Reply:
Now defined as `localized`.